wasm/libs/default_executor/src/multisig/account.rs

```diff
@@ -0,0 +1,96 @@
+//! Validation and execution logic of instructions for multisig accounts
+
+use super::*;
+
+impl VisitExecute for MultisigRegister {
+    fn visit(&self, executor: &mut Executor) {
+        let host = executor.host();
+        let target_domain = self.account.domain();
+
+        // Any account in a domain can register any multisig account in the domain
+        // TODO Restrict access to the multisig signatories?
+        // TODO Impose proposal and approval process?
+        if target_domain != executor.context().authority.domain() {
+            deny!(
+                executor,
+                "multisig account and its registrant must be in the same domain"
+            )
+        }
+
+        let Ok(_domain_found) = host
+            .query(FindDomains)
+            .filter_with(|domain| domain.id.eq(target_domain.clone()))
+            .execute_single()
+        else {
+            deny!(
+                executor,
+                "domain must exist before registering multisig account"
+            );
+        };
+
+        for signatory in self.signatories.keys().cloned() {
+            let Ok(_signatory_found) = host
+                .query(FindAccounts)
+                .filter_with(|account| account.id.eq(signatory))
+                .execute_single()
+            else {
+                deny!(
+                    executor,
+                    "signatories must exist before registering multisig account"
+                );
+            };
+        }
+    }
+
+    fn execute(self, executor: &mut Executor) -> Result<(), ValidationFail> {
+        let host = executor.host();
+        let registrant = executor.context().authority.clone();
+        let multisig_account = self.account;
+        let multisig_role = multisig_role_for(&multisig_account);
+
+        host.submit(&Register::account(Account::new(multisig_account.clone())))
+            .dbg_expect("registrant should successfully register a multisig account");
+
+        host.submit(&SetKeyValue::account(
+            multisig_account.clone(),
+            SIGNATORIES.parse().unwrap(),
+            Json::new(&self.signatories),
+        ))
+        .dbg_unwrap();
+
+        host.submit(&SetKeyValue::account(
+            multisig_account.clone(),
+            QUORUM.parse().unwrap(),
+            Json::new(&self.quorum),
+        ))
+        .dbg_unwrap();
+
+        host.submit(&SetKeyValue::account(
+            multisig_account.clone(),
+            TRANSACTION_TTL_MS.parse().unwrap(),
+            Json::new(&self.transaction_ttl_ms),
+        ))
+        .dbg_unwrap();
+
+        host.submit(&Register::role(
+            // Temporarily grant a multisig role to the registrant to delegate the role to the signatories
+            Role::new(multisig_role.clone(), registrant.clone()),
```

Comment:
```suggestion
            // Temporarily grant a multisig role to the registrant to delegate the role to the signatories
            Role::new(multisig_role.clone(), self.account.clone()),
```

yeah, this is a problem I also faced. The role should be given to the multisig account but then you can't execute grant instruction in the name of the multisig account. Since you just registered the account this must mean that you own the private key to that account. IMO, you should be able to execute instructions in the name of an account you created within a smart contract. 

Maybe we can model this with permissions? Whenever someone registers an account on someone's behalf they get the privileges to that account

Reply:
> Previously it was the design that the registrant of a multisig account got equivalent to the domain owner, so as to register and configure a multisig account in the domain. Now that host.submit() from executor goes through no validation, there's no need to elevate the authority. There should be essentially no difference between assigning any desired authority and bypassing any validation

this is not ok, registrant should be the domain owner in the default case (or have the corresponding permission). But if you just call the validation for the register instruction (and all other instructions) you will get this behavior

Reply:
that is, instead of calling `host.submit` you should call `executor.visit_XXX` for every instruction executed

Reply:
> Btw the concept of "keys of a multisig account" or "sign as a multisig account" will be removed by https://github.com/hyperledger-iroha/iroha/issues/5022 because keys should represent a personal identity

ok, disregard that part about providing signature

Reply:
Ok, regarding `MultisigRegister`, I have no objections to allow only who can `Register::account` to access for now, which would be just a specification change. If necessary, we can make a difference between the ability to register an account and the ability to organize existing accounts i.e. register a multisig account.

Regarding `MultisigPropose` `MultisigApprove`, I'm going to elevate the caller's authority to the multisig's one and execute via `visit_` functions. This internal temporally elevation is necessary since we cannot simply grant `CanModifyAccountMetadata { multisig }` to signatories